src/components/ha-camera-stream.ts

```diff
@@ -0,0 +1,145 @@
+
+    // Same entity, ignore.
+    if (curEntityId === oldEntityId) {
+      return;
+    }
+
+    // Tear down if we have something and we need to build it up
+    if (oldEntityId) {
+      this._teardownPlayback();
+    }
+
+    if (curEntityId) {
+      this._startPlayback();
+    }
+  }
+
+  private async _startPlayback(): Promise<void> {
+    if (!this.stateObj) {
+      return;
+    }
+
+    if (!this.hass!.config.components.includes("stream")) {
+      this._renderMJPEG();
+      return;
+    }
+
+    const videoEl = document.createElement("video");
+    videoEl.style.width = "100%";
+    videoEl.autoplay = true;
+    videoEl.controls = true;
+    videoEl.muted = true;
+    // @ts-ignore
+    videoEl.playsinline = true;
+
+    // tslint:disable-next-line
+    const Hls = ((await import(/* webpackChunkName: "hls.js" */ "hls.js")) as any)
+      .default as HLSModule;
+    let hlsSupported = Hls.isSupported();
+
+    if (!hlsSupported) {
+      hlsSupported =
+        videoEl.canPlayType("application/vnd.apple.mpegurl") !== "";
+    }
+
+    if (hlsSupported) {
+      try {
+        const { url } = await fetchStreamUrl(
+          this.hass!,
+          this.stateObj.entity_id
+        );
+
+        if (Hls.isSupported()) {
+          this._renderHLSPolyfill(videoEl, Hls, url);
+        } else {
+          this._renderHLSNative(videoEl, url);
+        }
+        return;
+      } catch (err) {
+        // Fails if entity doesn't support it. In that case we go
+        // for mjpeg.
+      }
+    }
+
+    this._renderMJPEG();
+  }
+
+  private async _renderHLSNative(videoEl: HTMLVideoElement, url: string) {
+    videoEl.src = url;
+    this.appendChild(videoEl);
+    await new Promise((resolve) =>
+      videoEl.addEventListener("loadedmetadata", resolve)
+    );
+    videoEl.play();
+  }
+
+  private async _renderHLSPolyfill(
+    videoEl: HTMLVideoElement,
+    // tslint:disable-next-line
+    Hls: HLSModule,
+    url: string
+  ) {
+    const hls = new Hls();
+    this._hlsPolyfillInstance = hls;
+    await new Promise((resolve) => {
+      hls.on(Hls.Events.MEDIA_ATTACHED, resolve);
+      hls.attachMedia(videoEl);
+    });
+    hls.loadSource(url);
+    this.appendChild(videoEl);
+    videoEl.addEventListener("loadeddata", () =>
+      fireEvent(this, "iron-resize")
+    );
+  }
+
+  private _renderMJPEG() {
+    const img = document.createElement("img");
+    img.style.width = "100%";
+    img.addEventListener("load", () => fireEvent(this, "iron-resize"));
+    img.src = __DEMO__
+      ? "/demo/webcamp.jpg"
+      : computeMJPEGStreamUrl(this.stateObj!);
+    img.alt = computeStateName(this.stateObj!);
+    this.appendChild(img);
+  }
+
+  private _teardownPlayback(): any {
+    if (this._hlsPolyfillInstance) {
+      this._hlsPolyfillInstance.destroy();
+      this._hlsPolyfillInstance = undefined;
+    }
+    while (this.lastChild) {
+      this.removeChild(this.lastChild);
+    }
+    this.stateObj = undefined;
```

Comment:
Few things here:

- This shouldn't be in tearDown, as that it also called during updating. That means that we change our entity ID during updating.
- The hack of using this, is also something that won't work when this component will be used in Lit, because Lit is too smart and won't set the property again when it re-renders a template that contains this component. That means that once detached, attaching it again won't rebuild up the video. This was different for the more info dialog, because I knew exactly how it was used.

The solution is that we should remove setting stateObj to undefined and in `connectedCallback`, we should trigger an update if no update is scheduled yet.

There is a private property that you can use, might have to ignore TypeScript: `this._hasRequestedUpdate` https://github.com/Polymer/lit-element/blob/master/src/lib/updating-element.ts#L662-L672